api/models/User.js

```diff
@@ -21,6 +27,69 @@ const schema = new mongoose.Schema({
 
 });
 
+/**
+ * Static methods.
+ */
+
+/**
+ * Query DS API for given User type/id and store.
+ * Using Promise.method -- http://stackoverflow.com/a/27844408/1470725.. but see below re: this.find
+ */
+UserSchema.statics.get = Promise.method((type, id) => {
+  logger.debug(`User.get type:${type} id:${id}`);
+
+  return app.locals.clients.northstar.Users
+    .get(type, id)
+    .then((northstarUser) => {
+      logger.debug('northstar.Users.get success');
+      const data = parseNorthstarUser(northstarUser);
+      console.log(this);
+      // return app.locals.db.users
+      // Per Mongoose documentaiton we should be able to use this.model('users')
+      // @see http://mongoosejs.com/docs/guide.html#methods
+      // That's why the this below in our instance method is breaking.
+      // return this.model('users')
+      return app.locals.db.users
+        .findOneAndUpdate({ _id: data._id }, data, { upsert: true, new: true, });
+    })
+    .catch((err) => {
+      logger.debug(`could not getUser type:${type} id:${id}`);
+      console.log(err);
+
+      return null;
+    });
+});
+
+/**
+ * Parse given Northstar User for User model.
+ */
+function parseNorthstarUser(northstarUser) {
+   const data = {
+    _id: northstarUser.id,
+    mobile: northstarUser.mobile,
+    first_name: northstarUser.firstName,
+    email: northstarUser.email,
+    phoenix_id: northstarUser.drupalID,
+    role: northstarUser.role,
+  };
+
+  return data;
+}
+
+/**
+ * Instance methods.
+ */
+
+ /**
+ * Set user's current_campaign and save signup on campaigns hash map.
+ */
+UserSchema.methods.setCurrentCampaignWithSignup = Promise.method((signup) => {
+  logger.debug(`setCurrentCampaignWithSignup signup:${signup.id}`);
+
+  console.log(this._id);
```

Comment:
I think it's undefined because it's called within `Promise.method`. Just declare it as a normal method and return `new Promise`, see [this](https://github.com/DoSomething/phoenix-js/blob/master/lib/phoenix-client.js#L44) for an example.